src/goto-programs/builtin_functions.cpp

```diff
@@ -28,8 +28,7 @@ static const std::string &get_string_constant(const exprt &expr)
     !expr.is_address_of() || expr.operands().size() != 1 ||
     !expr.op0().is_index() || expr.op0().operands().size() != 2)
   {
-    log_error("expected string constant, but got:\n{}", expr);
-    abort();
+    return {};
```

Comment:
Since the usage of this function is unexpected in this case, could we at least keep a warning? The missing name for the assertion is user-facing.